keel-core/src/main/kotlin/com/netflix/spinnaker/keel/constraints/DependsOnConstraintEvaluator.kt

```diff
@@ -0,0 +1,44 @@
+package com.netflix.spinnaker.keel.constraints
+
+import com.netflix.spinnaker.keel.api.DeliveryArtifact
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.api.DependsOnConstraint
+import com.netflix.spinnaker.keel.persistence.PromotionRepository
+import org.springframework.stereotype.Component
+
+@Component
+class DependsOnConstraintEvaluator(
```

Comment:
This is kinda the simplest possible constraint. It just checks if an artifact version is already in another (named) environment. Not was ever there, just is there now.

Reply:
I feel like eventually we'll need to support that an artifact was in another environment even if it isn't still (but that it wasn't rolled back out of that environment...) Thinking of a rapidly changing test/staging env and a slower changing prod env, you might want to not necessarily be promoting the thing that is currently there (maybe there is still some validation in flight) but rather something that was successfully there. Once you factor in some sort of manual approval constraint and deployment windows as well I could see this being a common thing.

Reply:
Definitely. I think this is the smallest possible step in that direction.

Reply:
It does this now